tests/unit/test_placeholders_with_steps.py

```diff
@@ -176,8 +176,22 @@ def test_step_input_order_validation():
 
 def test_map_state_with_placeholders():
     workflow_input = ExecutionInput()
```

Comment:
note: Placeholders don't have to be `ExecutionInput` all the time. You can use `StepInput` too

Reply:
I'll make sure to include both in next commit